homeassistant/components/systemmonitor/sensor.py

```diff
@@ -35,47 +35,83 @@
+        None,
+        False,
+    ],
+    "ipv4_address": ["IPv4 address", "", "mdi:server-network", None, True],
+    "ipv6_address": ["IPv6 address", "", "mdi:server-network", None, True],
+    "last_boot": ["Last boot", "", "mdi:clock", "timestamp", False],
+    "load_15m": ["Load (15m)", " ", CPU_ICON, None, False],
+    "load_1m": ["Load (1m)", " ", CPU_ICON, None, False],
+    "load_5m": ["Load (5m)", " ", CPU_ICON, None, False],
+    "memory_free": ["Memory free", DATA_MEBIBYTES, "mdi:memory", None, False],
+    "memory_use": ["Memory use", DATA_MEBIBYTES, "mdi:memory", None, False],
+    "memory_use_percent": [
+        "Memory use (percent)",
+        PERCENTAGE,
+        "mdi:memory",
+        None,
+        False,
+    ],
+    "network_in": ["Network in", DATA_MEBIBYTES, "mdi:server-network", None, True],
+    "network_out": ["Network out", DATA_MEBIBYTES, "mdi:server-network", None, True],
+    "packets_in": ["Packets in", " ", "mdi:server-network", None, True],
+    "packets_out": ["Packets out", " ", "mdi:server-network", None, True],
     "throughput_network_in": [
         "Network throughput in",
         DATA_RATE_MEGABYTES_PER_SECOND,
         "mdi:server-network",
         None,
+        True,
     ],
     "throughput_network_out": [
         "Network throughput out",
         DATA_RATE_MEGABYTES_PER_SECOND,
         "mdi:server-network",
+        True,
+    ],
+    "process": ["Process", " ", CPU_ICON, None, True],
+    "processor_use": ["Processor use", PERCENTAGE, CPU_ICON, None, False],
+    "processor_temperature": [
+        "Processor temperature",
+        TEMP_CELSIUS,
+        CPU_ICON,
         None,
+        False,
     ],
-    "process": ["Process", " ", CPU_ICON, None],
-    "processor_use": ["Processor use", PERCENTAGE, CPU_ICON, None],
-    "processor_temperature": ["Processor temperature", TEMP_CELSIUS, CPU_ICON, None],
-    "swap_free": ["Swap free", DATA_MEBIBYTES, "mdi:harddisk", None],
-    "swap_use": ["Swap use", DATA_MEBIBYTES, "mdi:harddisk", None],
-    "swap_use_percent": ["Swap use (percent)", PERCENTAGE, "mdi:harddisk", None],
+    "swap_free": ["Swap free", DATA_MEBIBYTES, "mdi:harddisk", None, True],
+    "swap_use": ["Swap use", DATA_MEBIBYTES, "mdi:harddisk", None, False],
+    "swap_use_percent": ["Swap use (percent)", PERCENTAGE, "mdi:harddisk", None, False],
 }
 
+
+def check_required_arg(value):
+    """Validate that the required "arg" for the sensor types that need it are set."""
+    for sensor in value:
+        type = sensor.get(CONF_TYPE)
+        arg = sensor.get(CONF_ARG)
+
+        if arg is None and SENSOR_TYPES[type][4] is True:
+            raise vol.RequiredFieldInvalid(
+                f"Mandatory 'arg' is missing for sensor type '{type}'."
+            )
+
+    return value
+
+
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_RESOURCES, default={CONF_TYPE: "disk_use"}): vol.All(
             cv.ensure_list,
+            check_required_arg,
```

Comment:
Put this validator after the list dict schema validator below. Then we know that the type key is present when we come to `check_required_arg`. `vol.All` evaluates the members sequentially.

Reply:
The reason being, we can then use direct `dict[key]` access for the "type" key which is a bit faster, right?

Reply:
Yeah, but for me it's more about doing things in the correct order. Each thing has its responsibility.